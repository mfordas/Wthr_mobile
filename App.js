import React from 'react';
import {
  SafeAreaView,
} from 'react-native';
import {createStackNavigator} from '@react-navigation/stack';
import {Provider} from 'react-redux';
import {NavigationContainer} from '@react-navigation/native';
import store from './src/store/store';
import MainScreen from './src/Components/MainScreen';
import Forecast from './src/Components/Forecast';

const Stack = createStackNavigator();

const App: () => React$Node = () => {
  return (
    <Provider store={store}>
      <NavigationContainer>
        <SafeAreaView style={{flex: 1}}>
          <Stack.Navigator>
            <Stack.Screen
              name="Main"
              component={MainScreen}
              options={{headerShown: false}}
            />
            <Stack.Screen
              name="Forecast"
              component={Forecast}
              options={{headerShown: false}}
            />
          </Stack.Navigator>
        </SafeAreaView>
      </NavigationContainer>
    </Provider>
  );
};

export default App;
