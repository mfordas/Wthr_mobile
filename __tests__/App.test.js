/**
 * @format
 */

import 'react-native';
import React from 'react';
import {Provider} from 'react-redux';
import App from '../App';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';

import store from '../src/store/store';

it('renders correctly', () => {
  const tree = renderer.create(<Provider store={store}><App /></Provider>).toJSON();
  expect(tree).toMatchSnapshot();
});
