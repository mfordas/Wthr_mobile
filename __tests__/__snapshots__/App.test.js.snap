// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <RNCSafeAreaView
      onInsetsChange={[Function]}
      style={
        Object {
          "flex": 1,
        }
      }
    >
      <View
        onLayout={[Function]}
        style={
          Object {
            "flex": 1,
          }
        }
      >
        <View
          pointerEvents="box-none"
          style={
            Object {
              "bottom": 0,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            }
          }
        >
          <View
            accessibilityElementsHidden={false}
            closing={false}
            gestureVelocityImpact={0.3}
            importantForAccessibility="auto"
            onClose={[Function]}
            onGestureBegin={[Function]}
            onGestureCanceled={[Function]}
            onOpen={[Function]}
            onTransitionStart={[Function]}
            pointerEvents="box-none"
            style={
              Object {
                "bottom": 0,
                "left": 0,
                "position": "absolute",
                "right": 0,
                "top": 0,
              }
            }
            transitionSpec={
              Object {
                "close": Object {
                  "animation": "spring",
                  "config": Object {
                    "damping": 500,
                    "mass": 3,
                    "overshootClamping": true,
                    "restDisplacementThreshold": 10,
                    "restSpeedThreshold": 10,
                    "stiffness": 1000,
                  },
                },
                "open": Object {
                  "animation": "spring",
                  "config": Object {
                    "damping": 500,
                    "mass": 3,
                    "overshootClamping": true,
                    "restDisplacementThreshold": 10,
                    "restSpeedThreshold": 10,
                    "stiffness": 1000,
                  },
                },
              }
            }
          >
            <View
              pointerEvents="box-none"
              style={
                Object {
                  "flex": 1,
                }
              }
            >
              <View
                enabled={false}
                hitSlop={
                  Object {
                    "right": -700,
                  }
                }
                maxDeltaY={20}
                minOffsetX={5}
                onHandlerStateChange={[Function]}
              >
                <View
                  style={
                    Object {
                      "flex": 1,
                      "transform": Array [
                        Object {
                          "translateX": 0,
                        },
                        Object {
                          "translateX": 0,
                        },
                      ],
                    }
                  }
                >
                  <View
                    pointerEvents="none"
                    style={
                      Object {
                        "backgroundColor": "rgb(242, 242, 242)",
                        "bottom": 0,
                        "left": 0,
                        "position": "absolute",
                        "shadowColor": "#000",
                        "shadowOffset": Object {
                          "height": 1,
                          "width": -1,
                        },
                        "shadowOpacity": 0.3,
                        "shadowRadius": 5,
                        "top": 0,
                        "width": 3,
                      }
                    }
                  />
                  <View
                    style={
                      Array [
                        Object {
                          "flex": 1,
                          "overflow": "hidden",
                        },
                        Array [
                          Object {
                            "backgroundColor": "rgb(242, 242, 242)",
                          },
                          undefined,
                        ],
                      ]
                    }
                  >
                    <View
                      style={
                        Object {
                          "flex": 1,
                          "flexDirection": "column-reverse",
                        }
                      }
                    >
                      <View
                        style={
                          Object {
                            "flex": 1,
                          }
                        }
                      >
                        <RCTScrollView
                          contentContainerStyle={
                            Object {
                              "flexGrow": 1,
                              "justifyContent": "space-between",
                            }
                          }
                          style={
                            Object {
                              "backgroundColor": "#03875b",
                              "flex": 1,
                            }
                          }
                        >
                          <View>
                            <View>
                              <View
                                accessible={true}
                                focusable={true}
                                onClick={[Function]}
                                onResponderGrant={[Function]}
                                onResponderMove={[Function]}
                                onResponderRelease={[Function]}
                                onResponderTerminate={[Function]}
                                onResponderTerminationRequest={[Function]}
                                onStartShouldSetResponder={[Function]}
                                style={
                                  Object {
                                    "alignContent": "center",
                                    "alignSelf": "center",
                                    "color": "black",
                                    "flexDirection": "row",
                                    "fontSize": 15,
                                    "height": 50,
                                    "justifyContent": "center",
                                    "margin": 5,
                                    "opacity": 1,
                                  }
                                }
                              >
                                <Image
                                  source={
                                    Object {
                                      "process": [Function],
                                    }
                                  }
                                  style={
                                    Object {
                                      "alignSelf": "center",
                                      "height": 20,
                                      "marginHorizontal": 10,
                                      "width": 20,
                                    }
                                  }
                                />
                                <Text
                                  style={
                                    Object {
                                      "alignSelf": "center",
                                      "fontFamily": "Exo2-VariableFont_wght",
                                      "fontSize": 20,
                                    }
                                  }
                                >
                                  City not found
                                  , 
                                </Text>
                                <Text
                                  style={
                                    Object {
                                      "alignSelf": "center",
                                      "fontFamily": "Exo2-VariableFont_wght",
                                      "fontSize": 20,
                                    }
                                  }
                                >
                                  try again
                                </Text>
                              </View>
                            </View>
                            <View
                              style={
                                Object {
                                  "alignItems": "center",
                                  "alignSelf": "center",
                                  "color": "black",
                                  "fontSize": 45,
                                  "justifyContent": "center",
                                  "margin": 0.5,
                                  "padding": 1,
                                  "textAlign": "center",
                                  "width": 200,
                                }
                              }
                            >
                              <Text
                                style={
                                  Object {
                                    "fontSize": 70,
                                  }
                                }
                              >
                                0
                                Â°C
                              </Text>
                              <View
                                style={
                                  Object {
                                    "alignItems": "center",
                                    "flexDirection": "row",
                                    "height": 50,
                                    "marginHorizontal": 5,
                                  }
                                }
                              >
                                <Image
                                  source={
                                    Object {
                                      "process": [Function],
                                    }
                                  }
                                  style={
                                    Object {
                                      "height": 40,
                                      "marginHorizontal": 15,
                                      "width": 40,
                                    }
                                  }
                                />
                                <Text
                                  style={
                                    Object {
                                      "fontFamily": "Exo2-VariableFont_wght",
                                      "fontSize": 20,
                                    }
                                  }
                                >
                                  0
                                   
                                  km/h
                                </Text>
                              </View>
                              <View
                                style={
                                  Object {
                                    "alignItems": "center",
                                    "flexDirection": "row",
                                    "height": 50,
                                    "marginHorizontal": 5,
                                  }
                                }
                              >
                                <Image
                                  source={
                                    Object {
                                      "process": [Function],
                                    }
                                  }
                                  style={
                                    Object {
                                      "height": 40,
                                      "marginHorizontal": 15,
                                      "width": 40,
                                    }
                                  }
                                />
                                <Text
                                  style={
                                    Object {
                                      "fontFamily": "Exo2-VariableFont_wght",
                                      "fontSize": 20,
                                    }
                                  }
                                >
                                  0
                                  %
                                </Text>
                              </View>
                              <View
                                style={
                                  Object {
                                    "alignItems": "center",
                                    "flexDirection": "row",
                                    "height": 50,
                                    "marginHorizontal": 5,
                                  }
                                }
                              >
                                <Image
                                  source={
                                    Object {
                                      "process": [Function],
                                    }
                                  }
                                  style={
                                    Object {
                                      "height": 40,
                                      "marginHorizontal": 15,
                                      "width": 40,
                                    }
                                  }
                                />
                                <Text
                                  style={
                                    Object {
                                      "fontFamily": "Exo2-VariableFont_wght",
                                      "fontSize": 20,
                                    }
                                  }
                                >
                                  0
                                   hPa
                                </Text>
                              </View>
                              <Image
                                source={
                                  Object {
                                    "process": [Function],
                                  }
                                }
                                style={
                                  Object {
                                    "height": 125,
                                    "margin": 15,
                                    "width": 125,
                                  }
                                }
                              />
                            </View>
                            <View
                              accessibilityRole="button"
                              accessibilityState={Object {}}
                              accessible={true}
                              focusable={true}
                              onClick={[Function]}
                              onResponderGrant={[Function]}
                              onResponderMove={[Function]}
                              onResponderRelease={[Function]}
                              onResponderTerminate={[Function]}
                              onResponderTerminationRequest={[Function]}
                              onStartShouldSetResponder={[Function]}
                              style={
                                Object {
                                  "opacity": 1,
                                }
                              }
                            >
                              <View
                                style={
                                  Array [
                                    Object {},
                                  ]
                                }
                              >
                                <Text
                                  style={
                                    Array [
                                      Object {
                                        "color": "#007AFF",
                                        "fontSize": 18,
                                        "margin": 8,
                                        "textAlign": "center",
                                      },
                                    ]
                                  }
                                >
                                  Go to 7 days forecast
                                </Text>
                              </View>
                            </View>
                            <View
                              style={
                                Object {
                                  "alignSelf": "center",
                                  "color": "black",
                                  "flexDirection": "row",
                                  "fontSize": 10,
                                  "margin": 0.5,
                                  "padding": 1,
                                  "textAlign": "center",
                                }
                              }
                            >
                              <Text>
                                Icons made by 
                              </Text>
                              <View
                                accessible={true}
                                focusable={true}
                                onClick={[Function]}
                                onResponderGrant={[Function]}
                                onResponderMove={[Function]}
                                onResponderRelease={[Function]}
                                onResponderTerminate={[Function]}
                                onResponderTerminationRequest={[Function]}
                                onStartShouldSetResponder={[Function]}
                                style={
                                  Object {
                                    "opacity": 1,
                                  }
                                }
                              >
                                <Text>
                                  Those Icons
                                </Text>
                              </View>
                              <Text>
                                 from 
                              </Text>
                              <View
                                accessible={true}
                                focusable={true}
                                onClick={[Function]}
                                onResponderGrant={[Function]}
                                onResponderMove={[Function]}
                                onResponderRelease={[Function]}
                                onResponderTerminate={[Function]}
                                onResponderTerminationRequest={[Function]}
                                onStartShouldSetResponder={[Function]}
                                style={
                                  Object {
                                    "opacity": 1,
                                  }
                                }
                              >
                                <Text>
                                  www.flaticon.com
                                </Text>
                              </View>
                            </View>
                          </View>
                        </RCTScrollView>
                      </View>
                    </View>
                  </View>
                </View>
              </View>
            </View>
          </View>
        </View>
      </View>
      <View
        pointerEvents="box-none"
        style={
          Object {
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          }
        }
      >
        <View
          accessibilityElementsHidden={false}
          importantForAccessibility="auto"
          onLayout={[Function]}
          pointerEvents="box-none"
          style={null}
        />
      </View>
    </RNCSafeAreaView>
  </View>
</RCTSafeAreaView>
`;
